compiler/test/test_end_to_end.re

```diff
@@ -664,6 +664,30 @@ let box_tests = [
   t("box_division2", "let b = box(76); b := unbox(b) / 19; ^b", "4"),
 ];
 
+let let_mut_tests = [
+  t("let-mut1", "let mut b = 4;b", "4"),
+  t("let-mut2", "let mut b = (4, (5, 6));b", "(4, (5, 6))"),
+  t("let-mut3", "let mut b = box(4);unbox(b)", "4"),
+  t("let-mut3_2", "let mut b = box(4); ^b", "4"),
+  t("let-mut4", "let mut b = 4;b = 3;b", "3"),
+  t("let-mut5", "let mut b = 4;b = b - 1;b", "3"),
+  tfile("counter-mut", "counter-mut", "1\n2\n3\nvoid"),
+  te(
+    "test_mut_typing",
+    "let mut a = false; a + 4",
+    "expression has type Bool but",
+  ),
+  /* Operations on Box<Number> */
+  t("let-mut_addition1", "let mut b = 4; b = b + 19", "23"),
+  t("let-mut_addition2", "let mut b = 4; b = b + 19; b", "23"),
+  t("let-mut_subtraction1", "let mut b = 4; b = b - 19", "-15"),
+  t("let-mut_subtraction2", "let mut b = 4; b = b - 19; b", "-15"),
+  t("let-mut_multiplication1", "let mut b = 4; b = b * 19", "76"),
+  t("let-mut_multiplication2", "let mut b = 4; b = b * 19; b", "76"),
+  t("let-mut_division1", "let mut b = 76; b = b / 19", "4"),
+  t("let-mut_division2", "let mut b = 76; b = b / 19; b", "4"),
```

Comment:
I re-implemented `+=`, etc., to work with mutable values rather than boxes. May be good to exercise those here too?